src/lib/cmdline.c

```diff
@@ -118,6 +118,141 @@ static int cbm_parse_cmdline_file_internal(const char *path, FILE *out)
         return nbytes;
 }
 
+/**
+ * Attempt to parse a command line file and remove its content from the out buffer.
+ *
+ * @Returns negative code if parsing failed, otherwise the new buffer size.
+ */
+static int cbm_parse_cmdline_file_removal_internal(const char *path, char *out, size_t buflen)
+{
+        autofree(FILE) *f = NULL;
+        size_t sn = 0;
+        ssize_t r = 0;
+        char *buf = NULL;
+        bool ret = true;
+        size_t nbytes = buflen;
+
+        f = fopen(path, "r");
+        if (!f) {
+                if (errno != ENOENT) {
+                        LOG_ERROR("Unable to open %s: %s", path, strerror(errno));
+                }
+                return false;
+        }
+
+        while ((r = getline(&buf, &sn, f)) > 0) {
+                ssize_t cur = 0;
+                char *m = NULL;
+                char *s = NULL;
+                size_t slen = 0;
+
+                /* Strip newlines */
+                if (r >= 1 && buf[r - 1] == '\n') {
+                        buf[r - 1] = '\0';
+                        --r;
+                }
+                char *l = buf;
+
+                /* Skip empty lines */
+                if (r < 1) {
+                        goto next_line;
+                }
+
+                /* Skip the starting whitespace */
+                while (isspace(*l)) {
+                        ++l;
+                        if (!*l) {
+                                break;
+                        }
+                }
+
+                /* Skip a comment */
+                if (l[0] == '#') {
+                        goto next_line;
+                }
+
+                /* Reset length */
+                cur = (l - buf);
+                r -= cur;
+
+                /* Strip trailing whitespace */
+                l = rstrip(l, (size_t)r, &r);
+
+                /* May now be an empty line */
+                if (r < 1) {
+                        goto next_line;
+                }
+
+                /* First look for a string with trailing whitespace */
+                s = string_printf("%s ", l);
```

Comment:
so maybe instead of doing this, check the next character after `memmem` matches?.. this way you'd get rid of "end of line" special case.